Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -164,6 +182,25 @@ + (void)clearConfiguration
     globalConfiguration() = nil;
 }
 
++ (void)maybeConsumeBundlePaths:(NSDictionary<NSAttributedStringDocumentReadingOptionKey, id> *)options
+{
+    auto* readAccessPaths = dynamic_objc_cast<NSArray<NSURL *>>(options[WKReadAccessFileURLsOption]);
+    if (!readAccessPaths)
+        return;
```

Comment:
Why is it correct to leave the old paths set when this is not specified? If a new path was specified, we’d overwrite the old ones, but unspecified just leaves the old ones alone? I think any time this is called, we should reset the old ones. I also think the method name should make the clear.

Or we can say it’s not legal to call this multiple times, in which case I think we should check that and not try to handle overwriting an old set with a new one.

Reply:
I didn't really consider that scenario, but you are correct -- it's a reasonable thing to do, and probably safer. One could imagine a case where someone needs to render an HTML string that requires access to some place we don't usually need access. Once that operation is compete, there is no need to continue using the cached WKWebView that happens to have that permission. Passing an empty set of directories should just tear down the old WKWebView and generate a "default" view that doesn't have those permissions.